airbyte-db/src/main/java/io/airbyte/db/bigquery/BigQueryDatabase.java

```diff
@@ -0,0 +1,225 @@
+    List<QueryParameterValue> parameterValueList;
+    if (params == null)
+      parameterValueList = Collections.emptyList();
+    else
+      parameterValueList = Arrays.stream(params).map(param -> QueryParameterValue.newBuilder().setValue(param).setType(
+          StandardSQLTypeName.STRING).build()).collect(Collectors.toList());
+
+    return query(sql, parameterValueList);
+  }
+
+  public Stream<JsonNode> query(String sql, List<QueryParameterValue> params) throws Exception {
+    final ImmutablePair<Job, String> result = executeQuery(bigQuery, getQueryConfig(sql, params));
+
+    if (result.getLeft() != null) {
+      FieldList fieldList = result.getLeft().getQueryResults().getSchema().getFields();
+      return Streams.stream(result.getLeft().getQueryResults().iterateAll())
+          .map(fieldValues -> BigQueryUtils.rowToJson(fieldValues, fieldList));
+    } else
+      throw new Exception(
+          "Failed to execute query " + sql + (params != null && !params.isEmpty() ? " with params " + params : "") + ". Error: " + result.getRight());
+  }
+
+  @Override
+  public void close() throws Exception {
+    /**
+     * The BigQuery doesn't require connection close. It will be done automatically.
+     */
+  }
+
+  public QueryJobConfiguration getQueryConfig(String sql, List<QueryParameterValue> params) {
+    return QueryJobConfiguration
+        .newBuilder(sql)
+        .setUseLegacySql(false)
+        .setPositionalParameters(params)
+        .build();
+  }
+
+  public ImmutablePair<Job, String> executeQuery(BigQuery bigquery, QueryJobConfiguration queryConfig) {
+    final JobId jobId = JobId.of(UUID.randomUUID().toString());
+    final Job queryJob = bigquery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
+    return executeQuery(queryJob);
+  }
+
+  /**
+   * Returns full information about all tables from entire project
+   *
+   * @param projectId BigQuery project id
+   * @return List of BigQuery tables
+   */
+  public List<Table> getProjectTables(String projectId) {
+    List<Table> tableList = new ArrayList<>();
+    bigQuery.listDatasets(projectId)
+        .iterateAll()
+        .forEach(dataset -> bigQuery.listTables(dataset.getDatasetId())
+            .iterateAll()
+            .forEach(table -> tableList.add(bigQuery.getTable(table.getTableId()))));
+    return tableList;
+  }
+
+  /**
+   * Returns full information about all tables from specific Dataset
+   *
+   * @param projectId BigQuery project id
+   * @param datasetId BigQuery dataset id
+   * @return List of BigQuery tables
+   */
+  public List<Table> getDatasetTables(String projectId, String datasetId) {
+    List<Table> tableList = new ArrayList<>();
+    bigQuery.listDatasets(projectId)
+        .iterateAll()
+        .forEach(dataset -> bigQuery.listTables(dataset.getDatasetId())
+            .iterateAll()
+            .forEach(table -> {
+              if (table.getTableId().getDataset().equalsIgnoreCase(datasetId))
```

Comment:
why do we do this check if we already list each table using `listTables(datasetId)`?

Reply:
Good point!